pkg/services/baremetal/host/host.go

```diff
@@ -1112,11 +1122,9 @@ func (s *Service) actionImageInstalling() actionResult {
 		return actionError{err: fmt.Errorf("failed to get running installimage processes: %q %q %w", out.StdOut, out.StdErr, out.Err)}
 	}
 	if out.StdOut != "" {
-		record.Warnf(s.scope.HetznerBareMetalHost, "InstallImageAlreadyRunning",
-			"installimage is already running:\n%s", out.StdOut)
-		return actionContinue{
-			delay: 10 * time.Second,
-		}
+		record.Warn(s.scope.HetznerBareMetalHost, "InstallImageAlreadyRunning",
+			"installimage is already running.")
+		return actionError{err: fmt.Errorf("installimage is already running")}
```

Comment:
why this change?

Reply:
If InstallImage is already running, then there is something strange. Running installimage again can work, but if the first installimage run failed, then the second try will fail with `ERROR unmounting device(s):`. See https://github.com/syself/cluster-api-provider-hetzner/pull/1407/files#diff-d1bdf47f722af9b77810d5839d1927cf67585d79b9f74cb5ce2c7586f4e9937eR265 

That's why I think it is better to start with getting into the rescue system again.

Reply:
but is it necessary at all, if you always return back to the registering state, to check for running install image processes at all? is there a case where this can still happen?

Reply:
@janiskemper if the controller restarts, then there can be an old install-image process still running.

Reply:
so if a running comand via ssh gets interrupted by a container restart? Or is it normal to start the operation (in the background) , then finish the control loop and start it again?

Reply:
> so if a running comand via ssh gets interrupted by a container restart?

Yes, then the install-image process on the bm-server will keep on running. The remote process won't notice that the ssh client has terminated. I added a comment to make this edge-case more obvious. It often happens via Tilt, but in production it can happen, too.

The Reconcile method call should terminate quickly. Installimage takes roughly two minutes. That's ok.

But if a process takes much longer, then we should handle it differently. Maybe via a Kubernetes-Job.

Reply:
Doing install-image async makes sense, I created a new issue for that: https://github.com/syself/cluster-api-provider-hetzner/issues/1408